### PR TITLE
feat(speak): add Flux TTS support via POST /v2/speak and the /v2/speak websocket

### PR DESCRIPTION
**Pull request by dg-coreylweathers**
## Summary

Adds `pkg/api/speak/v2` and `pkg/client/speak/v2`, mirroring the v1 rest/websocket split and the listen v2 (Flux STT) client patterns. Contracts follow the deepgram-docs AsyncAPI spec for the /v2/speak protocol and the OpenAPI spec for the batch endpoint — no invented wire fields.

**Batch REST (`POST /v2/speak`)**
- `ToStream` / `ToFile` / `ToSave` with `SpeakV2Options` (model required — no default, Aura models rejected; encoding, container, bit_rate, sample_rate, speed, expressivity, callback, callback_method, mip_opt_out, tag, priority)
- Callback (async) mode handled: the JSON acknowledgement `{"request_id":...}` arrives through the byte stream, per the spec, and is documented on the methods

**Streaming WebSocket (`wss://api.deepgram.com/v2/speak`)**
- Callback and channel clients with `Speak`, `Flush`, `Interrupt` / `InterruptWithOffset(ms)` (time_ms playback offset), `Configure` (speed), and graceful `Close`
- Binary audio frames routed to `Binary` handlers; all 10 typed server events (Connected, SpeechStarted, SpeechMetadata, SpeechInterrupted, Flushed, SessionMetadata, ConfigureSuccess, ConfigureFailure, Warning, Error)

## Testing

- `go build ./...` and `go test ./...` pass (golang:1.23 + portaudio)
- 24 new unit subtests: exact JSON for all client control messages, parsing for every server event (including optional interrupt text fields present/absent), URL building for both transports, and the REST transport end-to-end against a mock server (method/path/query/body/binary bytes/headers/400 propagation)
- golangci-lint v1.48.0 `--new-from-rev=main`: zero new findings
- **Live-verified** against production: batch returned a real MP3 with headers parsed; streaming produced a valid 48kHz PCM WAV through the full event lifecycle; `Configure(1.05)` → ConfigureSuccess and `Configure(3.5)` → ConfigureFailure `SPEED_OUT_OF_RANGE` with field/value parsed; `InterruptWithOffset(1500)` → SpeechInterrupted with the text_spoken/text_remaining split. The synthesized audio was transcribed verbatim by Flux STT (closed loop with #350)

Note for docs: the SDK validates explicitly supplied `Configure` speeds from 0.5 to 1.5 in 0.05 increments. An omitted speed is rejected as an empty update.

## Examples

- `examples/text-to-speech/rest/flux_batch` — synthesize a block of text to output.mp3
- `examples/text-to-speech/websocket/flux_streaming` — Speak/Flush one turn, save streamed linear16 as output.wav

Both read `DEEPGRAM_API_KEY`. All additive — `/v1/speak` and Aura are untouched.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/client/speak/v2/websocket/client_callback.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+		klog.V(7).Infof("fluxspeak.WSCallback.Configure() LEAVE\n")
+		return err
+	}
+
+	klog.V(4).Infof("Configure sent\n")
+	klog.V(7).Infof("fluxspeak.WSCallback.Configure() LEAVE\n")
+	return nil
+}
+
+// GetCloseMsg returns the JSON bytes for the Close control message. The server
```
Comment (GregHolmes) [inline pkg/client/speak/v2/websocket/client_callback.go:255]: **Title:** [B1] `Stop()` cannot provide the documented graceful close

**Summary:** Flux TTS `Close` drains active and queued turns, sends remaining audio, emits `SessionMetadata`, and then closes. The inherited `Stop()` sends `Close`, waits only 100 ms, sends a WebSocket close frame, and closes locally, so synthesis lasting longer than 100 ms can be cut off.

**Expected:** Closing waits for `SessionMetadata` and peer closure, bounded by a caller-visible timeout or context.

**Observed:** Transport shutdown starts after a fixed 100 ms regardless of queued synthesis, while the API docs promise a complete drain.

**Recommended fix:** Add a v2-specific graceful close that waits for `SessionMetadata` and server closure before closing locally, plus a separate immediate-abort path. Test with final audio and metadata delayed beyond 100 ms.

File: pkg/client/speak/v2/websocket/constants.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	// MessageTypeConfigure sends a mid-session configuration update
+	MessageTypeConfigure string = "Configure"
+
+	// MessageTypeClose gracefully closes the session
+	MessageTypeClose string = "Close"
+)
+
+const (
+	pingPeriod = 60 * time.Second
```
Comment (GregHolmes) [inline pkg/client/speak/v2/websocket/constants.go]: **Title:** [B2] Keepalive runs at the server timeout boundary

**Summary:** The server closes idle `/v2/speak` sessions after 60 seconds with `NET-0004`, while this ticker also waits 60 seconds before its first ping. Scheduler and network latency decide which wins.

**Expected:** Keepalive traffic reaches the server comfortably before the inactivity deadline.

**Observed:** The first ping is scheduled at the same boundary at which the server may close the session.

**Recommended fix:** Use a shorter interval such as 30 seconds and add a fake-clock or short-timeout test proving a ping occurs before the idle deadline.

File: examples/text-to-speech/websocket/flux_streaming/main.go [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+
+	sOptions := &interfaces.SpeakV2WSOptions{
+		Model:      model,
+		Encoding:   "linear16",
+		SampleRate: 48000,
+	}
+
+	// start the output file with a WAV container header so the raw linear16
+	// audio can be played with a standard media player. Sizes are left as
+	// placeholders, which most players accept for streamed WAV files.
```
Comment (GregHolmes) [inline examples/text-to-speech/websocket/flux_streaming/main.go]: **Title:** [B3] Streaming example writes an invalid WAV length declaration

**Summary:** RIFF/WAV requires the RIFF size at byte offset 4 and data length at byte offset 40. This example writes zero for both, appends audio, and never patches them; zero is not the standard streaming sentinel and tells conforming readers the chunks contain no audio.

**Expected:** The finished file declares `fileSize - 8` and the exact PCM data length.

**Observed:** Both fields remain `0x00000000` even after audio is appended.

**Recommended fix:** Keep one file handle open, count received bytes, seek back after `SpeechMetadata`, and patch both little-endian sizes before closing. Add a parser-based regression test.

File: pkg/api/speak/v2/rest/speak.go [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+
+	file, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o666)
+	if err != nil {
+		klog.V(1).Infof("os.OpenFile failed. Err: %v\n", err)
+		klog.V(6).Infof("speakv2.ToSave LEAVE\n")
+		return nil, err
+	}
+	defer file.Close()
+
+	result, err := c.ToFile(ctx, text, options, file)
```
Comment (GregHolmes) [inline pkg/api/speak/v2/rest/speak.go]: **Title:** [B4] Callback mode makes `ToSave` save JSON as audio

**Summary:** With `SpeakV2Options.Callback`, the response body is a request-ID acknowledgement rather than audio. `ToSave` still creates a file, delegates here, assigns `Filename`, and logs it as saved synthesis; the exported `AsyncResponse` is unused.

**Expected:** Callback requests return a typed asynchronous acknowledgement without creating a local audio file.

**Observed:** The JSON acknowledgement is written to the requested audio filename and reported as success.

**Recommended fix:** Reject callback options in `ToSave` and add a dedicated asynchronous method returning `AsyncResponse`. Test that callback mode never creates an audio file.

File: pkg/api/speak/v2/rest/speak.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+	klog.V(6).Infof("speakv2.ToFile LEAVE\n")
+
+	return result, err
+}
+
+// ToSave synthesizes text and saves the audio to the named file.
+func (c *Client) ToSave(ctx context.Context, filename, text string, options *interfaces.SpeakV2Options) (*api.SpeakResponse, error) {
+	klog.V(6).Infof("speakv2.ToSave ENTER\n")
+
+	file, err := os.OpenFile(filename, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o666)
```
Comment (GregHolmes) [inline pkg/api/speak/v2/rest/speak.go]: **Title:** [S1] `ToSave` destroys existing output before validation

**Summary:** `O_TRUNC` runs before `ToFile` validates the required model. Validation, request, writer, or close failures can erase an existing file or leave a partial artifact.

**Expected:** Invalid or failed requests leave an existing destination untouched.

**Observed:** The destination is truncated before local validation and remains empty or partial after later failures.

**Recommended fix:** Validate before opening. Prefer writing to a temporary sibling file, checking write/close errors, and atomically renaming only on success.

File: pkg/api/speak/v2/rest/speak.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+// ToStream synthesizes text and streams the audio into buf.
+// When options.Callback is set, the request is processed asynchronously and buf
+// receives the JSON acknowledgement {"request_id":"..."} instead of audio bytes.
+func (c *Client) ToStream(ctx context.Context, text string, options *interfaces.SpeakV2Options, buf *interfaces.RawResponse) (*api.SpeakResponse, error) {
+	klog.V(6).Infof("speakv2.ToStream ENTER\n")
+
+	keys := initializeKeys()
+
+	err := options.Check()
```
Comment (GregHolmes) [inline pkg/api/speak/v2/rest/speak.go]: **Title:** [S2] Nil inputs turn setup errors into panics

**Summary:** `ToStream` and `ToFile` call `options.Check()` without a nil guard; the WebSocket constructors and REST constructor have equivalent dereferences. `NewRESTWithDefaults` can also return nil when credentials are missing, which the example wraps and later dereferences after creating `output.mp3`.

**Expected:** Missing options or credentials return explicit errors without touching output files.

**Observed:** Plausible first-run mistakes produce nil-pointer panics.

**Recommended fix:** Add nil guards and expose an error-returning REST constructor. Until then, make the example detect a nil client before `ToSave`.

File: tests/unit_test/speak_flux_test.go [hunk 7]
```diff
@@ -0,0 +1,415 @@
...
+const (
+	testFluxModel  = "flux-haley-en"
+	testSpeechID   = "dg_sp_a1b2c3d4e5f6"
+	speakV2URLPath = "/v2/speak"
+	testAPIHost    = "api.deepgram.com"
+)
+
+// Test_FluxSpeakClientMessageSerialization verifies every client control message
+// serializes to the exact wire format from the /v2/speak AsyncAPI spec.
+func Test_FluxSpeakClientMessageSerialization(t *testing.T) {
```
Comment (GregHolmes) [inline tests/unit_test/speak_flux_test.go:38]: **Title:** [S3] WebSocket tests do not exercise WebSocket behavior

**Summary:** These tests marshal DTOs directly; the server tests similarly unmarshal DTOs. They never call either public client, either router, binary delivery, keepalive, or shutdown, so the suite passes despite the lifecycle blockers.

**Expected:** Both public clients are driven against a mock WebSocket with text/binary, routing, ordering, error, keepalive, and close assertions.

**Observed:** Only message struct serialization and parsing are covered.

**Recommended fix:** Add callback and channel mock-WebSocket tests for every outbound method and inbound event, and run the lifecycle suite under `-race`.

File: pkg/client/speak/v2/websocket/client_callback.go [hunk 8]
```diff
@@ -0,0 +1,260 @@
...
+}
+
+// Configure sends a mid-session configuration update to the server.
+// The server responds with ConfigureSuccess or ConfigureFailure.
+func (c *WSCallback) Configure(opts *clientinterfaces.SpeakV2ConfigureOptions) error {
+	klog.V(7).Infof("fluxspeak.WSCallback.Configure() ENTER\n")
+
+	msg := msginterfaces.ConfigureMessage{
+		Type:  MessageTypeConfigure,
+		Speed: opts.Speed,
```
Comment (GregHolmes) [inline pkg/client/speak/v2/websocket/client_callback.go:242]: **Title:** [S4] Configure cannot distinguish omitted speed from invalid zero

**Summary:** `Configure(nil)` panics, and `Speed: 0` is removed by `omitempty`, producing a fieldless Configure that can succeed as an empty update instead of rejecting the invalid value.

**Expected:** Nil returns an error and explicitly supplied zero is rejected.

**Observed:** Nil panics; zero is serialized as though omitted.

**Recommended fix:** Represent optional speed with a pointer or presence wrapper, validate nil input, and test omitted, zero, bounds, increments, and out-of-range values.

File: pkg/client/interfaces/v2/types-speak.go [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+	// "ogg" for opus, "none" for no container). Defaults per encoding.
+	Container string `json:"container,omitempty" schema:"container,omitempty"`
+	// SampleRate in Hz; supported values depend on Encoding. linear16: 8000, 16000,
+	// 24000, 32000, 44100, 48000. mulaw/alaw: 8000, 16000. flac: 8000, 16000, 22050,
+	// 32000, 48000. Not applicable to mp3 or opus. Defaults to the model's native rate.
+	SampleRate int `json:"sample_rate,omitempty" schema:"sample_rate,omitempty"`
+	// BitRate in bits per second for compressed encodings. mp3: 8000, 16000, 24000,
+	// 32000, 40000, 48000 (default). opus: 4000-650000. aac: 4000-192000.
+	BitRate int `json:"bit_rate,omitempty" schema:"bit_rate,omitempty"`
+	// Speed is the speech-rate multiplier: 0.85, 0.9, 0.95, 1.0 (default), 1.05, 1.1,
```
Comment (GregHolmes) [inline pkg/client/interfaces/v2/types-speak.go]: **Title:** [S5] Public speed documentation is stale

**Summary:** Production and the current OpenAPI, AsyncAPI, product docs, and changelog specify `0.5` to `1.5` in `0.05` increments. The REST, initial WebSocket, and Configure comments retain the obsolete `0.85` to `1.15` range.

**Expected:** Every public speed field documents the current range and increment.

**Observed:** The new Go surface advertises a narrower, obsolete contract.

**Recommended fix:** Update all three comments and the stale ConfigureFailure fixture text at `tests/unit_test/speak_flux_test.go:201`.

File: pkg/api/speak/v2/websocket/interfaces/types.go [hunk 10]
```diff
@@ -0,0 +1,249 @@
...
+	// TextSpoken is the portion of the turn's text the user heard. Omitted when the
+	// Interrupt carried no playback_offset.
+	TextSpoken string `json:"text_spoken,omitempty"`
+	// TextRemaining is the portion of the turn's text the user did not hear. Omitted
+	// when the Interrupt carried no playback_offset.
+	TextRemaining string       `json:"text_remaining,omitempty"`
+	Metadata      TurnMetadata `json:"metadata"`
+}
+
+// FlushedResponse is the immediate echo on receipt of a manual Flush, confirming
```
Comment (GregHolmes) [inline pkg/api/speak/v2/websocket/interfaces/types.go]: **Title:** [S6] `Flushed` timing is documented incorrectly

**Summary:** The protocol emits `Flushed` when the turn buffer has actually been flushed, and it may be held behind pending turns. It is not an immediate receipt echo.

**Expected:** Godoc describes buffer completion and identifies `SpeechMetadata` as the all-audio-delivered signal.

**Observed:** This comment gives developers an incorrect immediate ordering guarantee.

**Recommended fix:** Replace the immediate-echo wording here and in the callback interface with the published server semantics.

File: pkg/api/speak/v2/websocket/chan_router.go [hunk 11]
```diff
@@ -0,0 +1,377 @@
...
+
+	prettyjson "github.com/hokaccha/go-prettyjson"
+	klog "k8s.io/klog/v2"
+
+	interfaces "github.com/deepgram/deepgram-go-sdk/v3/pkg/api/speak/v2/websocket/interfaces"
+)
+
+// NewChanWithDefault creates a ChanRouter backed by the default channel handler.
+func NewChanWithDefault() *ChanRouter {
+	chans := NewDefaultChanHandler()
```
Comment (GregHolmes) [inline pkg/api/speak/v2/websocket/chan_router.go:23]: **Title:** [S7] Default channel helper starts duplicate consumers

**Summary:** `NewDefaultChanHandler()` already starts `Run()` automatically; this helper starts the same handler again, creating two consumer sets and leaving them blocked forever because the channels are never closed.

**Expected:** The default handler starts once and terminates its owned goroutines on shutdown.

**Observed:** This path starts duplicate consumers and has no shutdown mechanism.

**Recommended fix:** Remove the second `Run()` call, make startup ownership explicit, and add cancellation/channel closure with a goroutine-lifecycle test.

File: pkg/client/speak/v2/rest/client.go [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+	if err != nil {
+		klog.V(1).Infof("SetupRequest failed. Err: %v\n", err)
+		klog.V(6).Infof("speakv2.DoText() LEAVE\n")
+		return nil, err
+	}
+
+	// use HTTPClient + HandleResponse to extract the response headers alongside the body
+	var kv map[string]string
+	err = c.HTTPClient.Do(ctx, req, func(res *http.Response) error {
+		kv, err = c.HandleResponse(res, keys, resBody)
```
Comment (GregHolmes) [inline pkg/client/speak/v2/rest/client.go]: **Title:** [S8] Most HTTP errors lose their Deepgram details

**Summary:** The shared response handler reads and parses the body only for HTTP 400. Authentication failures, rate limits, payload limits, and 5xx responses return a bare `StatusError`; the HTTP layer then closes the unread body.

**Expected:** Every non-success response preserves typed or raw Deepgram error details.

**Observed:** Only HTTP 400 retains the response payload.

**Recommended fix:** Parse all non-success bodies and attach them to `StatusError`; add 401, 403, 413, 429, and 500 fixtures.

**Review by GregHolmes — CHANGES_REQUESTED** [review]
Requesting changes: 4 blocking findings and 8 should-fix findings. The Go 1.19 test, race, vet, changed-file lint, and Markdown gates pass, but the current tests do not exercise the new WebSocket clients or shutdown lifecycle. The blocking issues are graceful close truncating queued audio, keepalive racing the server timeout, invalid WAV size fields in the streaming example, and callback-mode `ToSave` reporting JSON as saved audio.

File: pkg/client/speak/v2/websocket/client_callback.go (hunk 1 above)
Comment (dg-coreylweathers) [reply pkg/client/speak/v2/websocket/client_callback.go:255]: Fixed in c8992cf. Both WebSocket clients now have a v2-specific graceful close, `Finish(ctx)`: it sends `Close`, then waits — bounded by the caller's ctx — for the final `SessionMetadata` **and** the server's own socket closure before tearing anything down, so queued synthesis is never cut off by the old fixed 100ms window. `Stop()` remains as the separate immediate-abort path, and a peer close without `SessionMetadata` returns `ErrGracefulCloseIncomplete`.

Tested exactly as requested: the mock server delays the turn's final audio + metadata 300ms past the old 100ms window and both clients' tests assert the delayed audio and `SessionMetadata` were delivered before `Finish` returned (`tests/unit_test/speak_flux_ws_test.go`, incl. timeout and close-without-metadata edge cases). Live-verified too: a 28-second turn (2.7MB of PCM) drained completely through `Finish(ctx)` against api.deepgram.com.

File: pkg/client/speak/v2/websocket/constants.go (hunk 2 above)
Comment (dg-coreylweathers) [reply pkg/client/speak/v2/websocket/constants.go]: Fixed in c8992cf. The keepalive default is now 30 seconds (`DefaultKeepAlivePeriod`) — half the server's 60s NET-0004 idle deadline — instead of racing it at the boundary, and `ClientOptions.KeepAlivePeriod` makes the interval configurable. Tests: a static assertion that the default is at most half of `ServerIdleTimeout`, plus a short-timeout mock-server test (50ms interval) proving protocol pings actually arrive on the configured interval (`Test_FluxSpeakKeepAlive`).

File: examples/text-to-speech/websocket/flux_streaming/main.go (hunk 3 above)
Comment (dg-coreylweathers) [reply examples/text-to-speech/websocket/flux_streaming/main.go]: Fixed in c8992cf. The example now streams audio through a new minimal streaming WAV writer (`pkg/audio/wav`): one file handle for the whole session, byte counting on every write, and `Finalize()` seeking back to patch both little-endian fields — RIFF size (`fileSize-8`) at offset 4 and the exact PCM data length at offset 40 — after the session drains.

Parser-based regression test included (`Test_WAVWriterDeclaresRealSizes`): asserts both fields at their exact offsets and round-trips the file through the `youpy/go-wav` parser. Live run verified the produced `output.wav` (RIFF size == fileSize-8, data == fileSize-44; Python's `wave` module parses it as mono/48kHz/16-bit, 28.0s).

File: pkg/api/speak/v2/rest/speak.go (hunk 4 above)
Comment (dg-coreylweathers) [reply pkg/api/speak/v2/rest/speak.go]: Fixed in c8992cf. `ToSave` now rejects callback options with `ErrCallbackNotSupported` before any file or network I/O, and the new `ToAsync(ctx, text, options)` returns the typed `api.AsyncResponse` (which is now actually used). `Test_FluxSpeakToSaveCallbackRejected` asserts callback mode never creates a local file (and never even sends a request from ToSave), and that `ToAsync` returns the acknowledged `request_id` / requires a callback URL.

File: pkg/api/speak/v2/rest/speak.go (hunk 5 above)
Comment (dg-coreylweathers) [reply pkg/api/speak/v2/rest/speak.go]: Fixed in c8992cf. `ToSave` validates everything (nil guards, `Check()`, callback rejection) before touching the destination, then writes to a temporary sibling file (`os.CreateTemp` in the same directory), checks write/close errors, and atomically renames onto the destination only on success — failures remove the temp file and leave any existing destination byte-for-byte intact. Covered by `Test_FluxSpeakToSavePreservesExistingFile` (validation failure, HTTP 500 failure, and the atomic-success case, each asserting no temp files are left behind).

File: pkg/api/speak/v2/rest/speak.go (hunk 6 above)
Comment (dg-coreylweathers) [reply pkg/api/speak/v2/rest/speak.go]: Fixed in c8992cf. `ToStream`/`ToFile`/`ToSave` now return `ErrOptionsRequired` on nil options and `ErrNilClient` on a nil transport; the WebSocket constructors take nil `cOptions` (defaulted) and reject nil `sOptions` with an error. Since the package is unreleased, I made the REST constructors themselves error-returning (`New` / `NewWithDefaults` return `(*Client, error)`) rather than adding a parallel constructor — a nil client can no longer escape, and the batch example checks the error before use. Covered by `Test_FluxSpeakNilGuards`.

File: tests/unit_test/speak_flux_test.go (hunk 7 above)
Comment (dg-coreylweathers) [reply tests/unit_test/speak_flux_test.go:38]: Fixed in c8992cf. `tests/unit_test/speak_flux_ws_test.go` now drives **both public clients** end to end against a mock TLS WebSocket server: every outbound method (`Speak`, `Flush`, `Interrupt`, `InterruptWithOffset`, `Configure`) asserted against its exact wire frame; inbound routing for text events and binary audio, including ordering (audio delivered before the turn's metadata); keepalive pings; the full shutdown lifecycle (graceful `Finish` drain, ctx timeout, peer-close-without-metadata); and post-shutdown write-error propagation. The whole suite runs under `-race` (verified on Go 1.23 and CI's Go 1.19).

File: pkg/client/speak/v2/websocket/client_callback.go (hunk 8 above)
Comment (dg-coreylweathers) [reply pkg/client/speak/v2/websocket/client_callback.go:242]: Fixed in c8992cf. `SpeakV2ConfigureOptions.Speed` is now `*float64` (as is `ConfigureMessage.Speed`), so omitted and explicit-zero are distinguishable: `Configure(nil)` returns `ErrOptionsRequired`, a fieldless Configure returns `ErrConfigureNoFields` instead of sneaking through as an empty update, and a set value is validated client-side against the current contract (0.5–1.5 in 0.05 increments) so an explicit 0 is rejected with `ErrSpeedOutOfRange`. `Test_FluxSpeakConfigureValidation` covers omitted, zero, bounds (0.5/1.5 accepted, 0.45/1.55 rejected), increments (1.07 rejected), and out-of-range values; the serialization tests additionally pin that a set zero serializes `{"speed":0}` and nil omits the field.

File: pkg/client/interfaces/v2/types-speak.go (hunk 9 above)
Comment (dg-coreylweathers) [reply pkg/client/interfaces/v2/types-speak.go]: Fixed in c8992cf. All three public speed comments (REST `SpeakV2Options.Speed`, WebSocket `SpeakV2WSOptions.Speed`, and `SpeakV2ConfigureOptions.Speed`) now document 0.5–1.5 in 0.05 increments, and the stale `ConfigureFailure` fixture text in `tests/unit_test/speak_flux_test.go` was updated to match.

File: pkg/api/speak/v2/websocket/interfaces/types.go (hunk 10 above)
Comment (dg-coreylweathers) [reply pkg/api/speak/v2/websocket/interfaces/types.go]: Fixed in c8992cf. The `FlushedResponse` godoc and both handler-interface comments now describe the published semantics: `Flushed` is emitted when the turn's buffer has actually been flushed, may be held behind turns still synthesizing (so it is not an immediate receipt echo), and `SpeechMetadata` is the all-audio-delivered signal for the turn.

File: pkg/api/speak/v2/websocket/chan_router.go (hunk 11 above)
Comment (dg-coreylweathers) [reply pkg/api/speak/v2/websocket/chan_router.go:23]: Fixed in c8992cf. `NewDefaultChanHandler` no longer starts `Run()` itself — startup ownership is explicit at the call sites (`NewChanWithDefault` and the client factory's nil-handler path each start it exactly once), so duplicate consumer sets can no longer be created. The new `Shutdown()` closes every owned channel, terminating the consumer goroutines and letting `Run` return; `Test_FluxSpeakDefaultChanHandlerLifecycle` proves the constructor starts nothing, a single `Run` consumes, and `Shutdown` ends it (idempotently).

File: pkg/client/speak/v2/rest/client.go (hunk 12 above)
Comment (dg-coreylweathers) [reply pkg/client/speak/v2/rest/client.go]: Fixed in c8992cf. The speak v2 REST path now parses the body of **every** non-success response — not just HTTP 400 — into a `StatusError` carrying the typed `DeepgramError` (falling back to the raw body text when it isn't the Deepgram JSON shape). `Test_FluxSpeakRESTErrorDetails` adds the requested 401, 403, 413, 429, and 500 fixtures plus a non-JSON 502 case. Also live-verified: the real /v2/speak 401 body (`{"err_code":"INVALID_AUTH","err_msg":"Invalid credentials.",...}`) parses into the typed error.

**Comment by dg-coreylweathers** [comment]
All 12 findings (B1–B4, S1–S8) are addressed in c8992cf — I replied on each inline thread with what changed and the tests behind it.

Beyond the direct fixes, the same commit hardens a few things a second review pass surfaced:

- **Reconnect-safe Finish**: the graceful-close milestones are per-session, so `Finish(ctx)` after `AttemptReconnect` waits for the *new* session's drain instead of returning instantly on the old session's already-fired milestones (regression test: `Test_FluxSpeakFinishAfterReconnect`). `Start()` on reconnect can also no longer stack a second keepalive goroutine.
- **`pkg/audio/wav` is documented as linear16/PCM-only** (mulaw/alaw would need format codes 7/6), with a matching note in the example README.
- **The streaming example fails loudly on an empty session** (removes the file and exits non-zero instead of leaving a 44-byte header-only WAV).

Live verification (run from Docker against api.deepgram.com):
- Batch: real MP3 saved via `ToSave`, response headers parsed (request-id, model, char-count).
- Streaming: a 28-second turn (2.7MB of PCM) drained completely through `Finish(ctx)`; the produced `output.wav` declares RIFF size = fileSize−8 (offset 4) and the exact PCM length (offset 40), and parses in standard WAV tooling.
- Errors: the live `/v2/speak` 401 body (`{"err_code":"INVALID_AUTH","err_msg":...}`) parses into the typed `StatusError`.

Gates on the final state (Docker): `go build`, `go vet`, `go test -race` on Go 1.23 and CI's Go 1.19, `golangci-lint v1.48.0 --new-from-rev=main` — all clean.

**Review by GregHolmes — DISMISSED** [review]

**Comment by dg-coreylweathers** [comment]
Sorry to bounce this back to you, Greg — GitHub dismissed your approval on push. **The re-review is four lines.**

Since your approval at `5139f1f`, the PR's own content changed by exactly this:

```
-func TestRunWithoutCredentialsPreservesExistingOutput(t *testing.T) {
+func Test_RunWithoutCredentialsPreservesExistingOutput(t *testing.T) {
-func TestTemporaryOutputPreservesOrReplacesDestination(t *testing.T) {
+func Test_TemporaryOutputPreservesOrReplacesDestination(t *testing.T) {
-func TestKeepaliveStateReplacesAndStopsPingers(t *testing.T) {
+func Test_KeepaliveStateReplacesAndStopsPingers(t *testing.T) {
+func Test_WSChannelDefaultHandlerStopsWithLifecycle(t *testing.T) {
```

To see it yourself, ignoring the `main` merge:

```bash
git diff 5139f1f 3cbc94e -- pkg/client/speak/v2 examples/text-to-speech/websocket/flux_streaming
```

**Why the renames:** CI runs `go test -v -run Test_ ./...`, so without the prefix these two packages reported `[no tests to run]` and never executed — including `Test_KeepaliveStateReplacesAndStopsPingers`, which covers the per-session keepalive rework in `5139f1f`. CLAUDE.md requires the prefix for exactly this reason. Verified before and after with the real CI command: both packages went from `[no tests to run]` to executing, and all four tests plus five subtests pass.

The other commit here (`f2ea203`) is a merge of `origin/main` — zero conflicts, disjoint files (`main` touched only speak v1 and docs; this branch is speak v2, `pkg/audio/wav` and `pkg/client/common`). No behavior change. I re-verified after the merge that the `Finish(ctx)` shim, reconnect-safe finish, the `keepaliveState` pinger, the WAV writer and the `dupl` exclusions all survived intact, and `Test_FluxSpeakFinishAfterReconnect` still passes.

CI is green on `3cbc94e`. Nothing else changed.

Also still needs a second approval from @deepgram-kiley under the unattributed-changes rule.

**Review by GregHolmes — DISMISSED** [review]

**Review by GregHolmes — APPROVED** [review]
